src/main/java/com/fasterxml/jackson/core/StreamReadConfig.java

```diff
@@ -0,0 +1,28 @@
+package com.fasterxml.jackson.core;
+
+/**
+ * The configuration for streaming reads.
+ *
+ * @since 2.15
+ */
+public class StreamReadConfig {
+
+    private static final int DEFAULT_MAX_NUM_LEN = 1000;
+
+    private int _maxNumLen = DEFAULT_MAX_NUM_LEN;
+
+    /**
+     * Sets the maximum number length (in chars). The default is 1000 (since Jackson 2.14)
+     * @param maxNumLen the maximum number length (in chars)
+     * @return this config
+     * @since 2.15
+     */
+    public StreamReadConfig maxNumberLength(int maxNumLen) {
```

Comment:
I think these must be immutable, so changes via `withXxx()` methods, or using full builders. Mutable instances are not thread-safe and would require more copying.